oeps/oep-0031-arch-i18n.rst

```diff
@@ -0,0 +1,212 @@
+
+Not Everything Is English
+=========================
+
+Use longer strings with placeholder variables, rather than trying to create messages by concatenating two or more short messages together. Even if it looks like it's working in English, it won't necessarily be grammatically correct in other languages. This includes uses such as a pagination display: you cannot translate only the word ``of`` and then glue together ``{number} + {translation of "of"} + {number}``. Instead, your string for translation should be ``{page_number} of {total_pages}``, so your translators can move the placeholders into whatever position is required for their language.
+
+When creating forms, keep in mind that there are many different ways of writing an address besides "street number and street name" as one field, and city and state as two others. Also, not every culture puts given name and family name in that order. If possible, allow free form, unstructured addresses, name fields, etc.
+
+Accessibility
+=============
+
+Translate all non-display text that is present in a tag for accessibility reasons (e.g., ``aria-label``), using the same locale as the rest of the page.
+
+Dates And Numbers
+=================
+
+All dates and numbers should be localized.
+
+Currency
+========
+
+When displaying a price in a foreign (to the user) currency, localize the number according to the user's locale. That is, for a user in France viewing a price in U.S. dollars, show it as ``USD 19,99``, using the ``fr`` convention for the decimal point.
+
+Right-To-Left Locales
+=====================
+
+For languages that are written right-to-left (RTL), such as Arabic and Hebrew, you will need to add a ``dir`` attribute to your ``html`` tag, and your CSS must swap graphics and margins. The frontend-i18n library will make this straightforward; see the `i18n HOWTO <https://github.com/edx/frontend-i18n/blob/master/docs/how_tos/i18n.rst>`_ for more information.
+
+*****************
+Locale Resolution
+*****************
+
+For pages meant to be seen by logged-in users, we will check the user's cookies for a locale; if there isn't one, we will fetch it from the user's account settings and store it in their cookies for reuse. If there is no locale (or no locale we support) in the account settings, we will use the locale that the user's browser is set to. Failing everything else, we will fall back to English.
+
+Alternatives Considered
+=======================
+
+We considered but rejected a fallback step where we would choose a language based on the geographic location of the user's IP address. This is generally not reliable: many countries have speakers of multiple languages, users might be traveling, and users might be living or studying in a country but still be more comfortable in a different language.
+
+****
+URLs
```

Comment:
To distinguish from the `Locale Resolution` section, consider renaming this section to `URLs and Locale Resolution for public pages`.